app/src/ui/diff/text-diff.tsx

```diff
@@ -1019,6 +1027,28 @@ export class TextDiff extends React.Component<ITextDiffProps, ITextDiffState> {
       : canSelect(file) && file.selection.isSelected(index)
   }
 
+  /**
+   * Returns a string with the pixel size for the gutter width.
+   * 1 = 20
+   * >1 = (10 * digit amount) + 5
+   */
+  private getGutterLineWidth(totalHunks: readonly DiffHunk[]): number {
+    // Get the largest old line number and the largest new line, of these two we can find the highest amount of digits in the hunks
+    const largestOldLineNum = Math.max(
+      ...totalHunks.map(function (o) {
+        return o.lines[o.lines.length - 1].oldLineNumber ?? 0
+      })
+    )
+
+    const largestNewLineNum = Math.max(
+      ...totalHunks.map(function (o) {
+        return o.lines[o.lines.length - 1].newLineNumber ?? 0
+      })
+    )
```

Comment:
As mentioned above, this way of obtaining the largest line numbers seems it could be inefficient, and just look backwards from the end of the diff should work.

Another option that might be more efficient is to carry that information in the `ITextDiffData` interface, and obtain it when the diff is parsed in https://github.com/desktop/desktop/blob/21e071f5eff5e533afe4d4ef39ef7d1e4d7e9f9b/app/src/lib/diff-parser.ts#L386 and then keep it updated when the user expands the diff in https://github.com/desktop/desktop/blob/21e071f5eff5e533afe4d4ef39ef7d1e4d7e9f9b/app/src/ui/diff/text-diff-expansion.ts#L173

That way, this info has always been pre-calculated and available.

What do you think?

Reply:
That sounds a lot better since I believe this is ran several times when the view is refreshed, but I wasn't sure of a better place.

A long with the render method suggestion I'll try and scoot stuff around and get it working.

Reply:
This works excellently and is much better than calculating it on every single render of the page. Thank you!